base/poco/MongoDB/src/Connection.cpp

```diff
@@ -145,68 +145,155 @@ void Connection::connect(const Poco::Net::StreamSocket& socket)
+        {
+            tempstr += *it;
+            tail++;
+        }
+
+        it = tempstr.begin();
+        std::string token;
+        for (;it != tempstr.end(); ++it)
+        {
+            if (*it == ',')
+            {
+                newURI = uri.substr(0, head) + token + uri.substr(tail, uri.length());
+                strAddresses.push_back(newURI);
+                token = "";
+            }
+            else
+            {
+                token += *it;
+            }
+        }
+        newURI = uri.substr(0, head) + token + uri.substr(tail, uri.length());
+        strAddresses.push_back(newURI);
+    }
+    else
+    {
+        strAddresses.push_back(uri);
+    }
+
+    newURI = strAddresses.front();
+    Poco::URI theURI(newURI);
+    if (theURI.getScheme() != "mongodb") throw Poco::UnknownURISchemeException(uri);
+
+    std::string userInfo = theURI.getUserInfo();
+    std::string databaseName = theURI.getPath();
+    if (!databaseName.empty() && databaseName[0] == '/') databaseName.erase(0, 1);
+    if (databaseName.empty()) databaseName = "admin";
+
+    bool ssl = false;
+    Poco::Timespan connectTimeout;
+    Poco::Timespan socketTimeout;
+    std::string authMechanism = Database::AUTH_SCRAM_SHA1;
+    std::string readPreference="primary";
+
+    Poco::URI::QueryParameters params = theURI.getQueryParameters();
+    for (Poco::URI::QueryParameters::const_iterator it = params.begin(); it != params.end(); ++it)
+    {
+        if (it->first == "ssl")
+        {
+            ssl = (it->second == "true");
+        }
+        else if (it->first == "connectTimeoutMS")
+        {
+            connectTimeout = static_cast<Poco::Timespan::TimeDiff>(1000)*Poco::NumberParser::parse(it->second);
+        }
+        else if (it->first == "socketTimeoutMS")
+        {
+            socketTimeout = static_cast<Poco::Timespan::TimeDiff>(1000)*Poco::NumberParser::parse(it->second);
+        }
+        else if (it->first == "authMechanism")
+        {
+            authMechanism = it->second;
+        }
+        else if (it->first == "readPreference")
+        {
+            readPreference= it->second;
+        }
+    }
+
+    for (std::vector<std::string>::const_iterator it = strAddresses.cbegin();it != strAddresses.cend(); ++it)
+    {
+        newURI = *it;
+        theURI = Poco::URI(newURI);
+
+        std::string host = theURI.getHost();
+        Poco::UInt16 port = theURI.getPort();
+        if (port == 0) port = 27017;
+
+        connect(socketFactory.createSocket(host, port, connectTimeout, ssl));
+        _uri = newURI;
+        if (socketTimeout > 0)
+        {
+            _socket.setSendTimeout(socketTimeout);
+            _socket.setReceiveTimeout(socketTimeout);
+        }
+        if (strAddresses.size() > 1)
+        {
+            Poco::MongoDB::QueryRequest request("admin.$cmd");
+            request.setNumberToReturn(1);
+            request.selector().add("isMaster", 1);
```

Comment:
is it correct that here you use `isMaster` and below `ismaster`?

Reply:
> is it correct that here you use `isMaster` and below `ismaster`?


Yes, isMaster is a command that returns the document containing ismaster as a boolean. More about it here: https://www.mongodb.com/docs/v4.0/reference/command/isMaster/